File: src/main/java/org/app/infrastructure/repositories/UserRepository.java
package org.app.infrastructure.repositories;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.app.domain.User;
import org.app.domain.IUserRepository;
import org.app.infrastructure.local.inmemory.InMemoryUserRepository;
import org.app.infrastructure.local.mongodb.MongoDBUserRepository;

import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class UserRepository implements IUserRepository {
    @Inject
    InMemoryUserRepository userRepository;

    @Override
    public void register(User user) {
        userRepository.register(user);
    }

    @Override
    public List<User> find(UUID userId) {
        return userRepository.find(userId);
    }

    @Override
    public List<User> find(List<UUID> userIds) {
        return userRepository.find(userIds);
    }

    @Override
    public List<User> findByUsername(String username) {
        return userRepository.findByUsername(username);
    }
}

File: go/app/main/infrastructure/repositories/user_repository.go
package repositories

import (
    "github.com/google/uuid"
    "mutualAttentionSystem/app/main/domain"
)

type UserRepository struct {
    access domain.IUserRepository
}

func NewUserRepository(access domain.IUserRepository) domain.IUserRepository {
    return &UserRepository{access}
}

func (repository *UserRepository) Find(id uuid.UUID) *domain.User {
    return repository.access.Find(id)
}

func (repository *UserRepository) FindByIds(ids []uuid.UUID) []*domain.User {
    return repository.access.FindByIds(ids)
}

func (repository *UserRepository) FindByUsername(username string) *domain.User {
    return repository.access.FindByUsername(username)
}

func (repository *UserRepository) Save(user *domain.User) {
    repository.access.Save(user)
}
